src/main/java/com/infoshareacademy/service/RecipeService.java

```diff
@@ -123,4 +134,47 @@ public void deleteRecipeById(Long id) {
         recipeRepository.deleteById(id);
     }
 
+    public void deleteAllRecipes() {
+        recipeRepository.deleteAll();
+    }
+
+    public Page<Recipe> getRecipeByProductsInFridge(Pageable pageable) {
+
+        //TODO: zmienić wczytywanie przepisów po ID
+        List<Recipe> myRecipes = getAllRecipe().stream()
+                .peek(recipe -> recipe.getProductList()
+                        .forEach(productRecipe -> productRecipe
+                                .setProductName(productRecipe.getProductName().toLowerCase()))).toList();
+
+        Map<String, ProductInFridge> productsInFridge = fridgeService.getAllProductsFromFridge()
+                .getProductsInFridge()
+                .stream()
+                .peek(productInFridge -> productInFridge
+                        .setProductName(productInFridge.getProductName().toLowerCase()))
+                .collect(Collectors.toMap(ProductInFridge::getProductName, Function.identity()));
+        List<Recipe> filteredRecipies = new ArrayList<>();
+
+
+        Map<Recipe, List<ProductRecipe>> mapRecipesProducts = myRecipes.stream()
+                .collect(Collectors.toMap(Function.identity(), Recipe::getProductList));
+
+        for (Map.Entry<Recipe, List<ProductRecipe>> entry : mapRecipesProducts.entrySet()) {
+            List<ProductRecipe> tempRecipeList = entry.getValue();
+            int matchScore = tempRecipeList.size();
+            for (ProductRecipe productRecipe : tempRecipeList) {
+                ProductInFridge productInFridge = productsInFridge.get(productRecipe.getProductName());
+
+                if (productInFridge != null
+                        && productInFridge.getAmount() >= productRecipe.getAmount()) {
+                    matchScore--;
+                }
+            }
+            if (matchScore == 0) {
+                filteredRecipies.add(entry.getKey());
+            }
+        }
+
+        return new PageImpl<>(filteredRecipies);
+    }
```

Comment:
To jest zdecydowanie do rozbicia. 
getRecipes
getProductsFridge
mapRecipesProducts